.github/workflows/gobuild.yml

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (C) 2015 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Go Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    name: Kubernetes Model Generator Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build Project
+        run: |
+          ./kubernetes-model-generator/generateModelDocker.sh
+          git status
+          MODIFIED_KUBE_SCHEMA=`git status | grep kube-schema.json | wc -l`
+          if [ $MODIFIED_KUBE_SCHEMA -eq "0" ]; then
+            echo "No Schema files modified. Build Success.";
+          else
+            echo "Schema files modified. Build Failure";
+            exit 1;
+          fi;
```

Comment:
The verification process should be added in a different step.

This way we can actually check if the go modules run fine (1).

And if the generated schemas have changed (change in go modules) but haven't been persisted in the VCS (2).